src/lib.rs

```diff
@@ -152,418 +152,19 @@
-        self.level
-    }
-
-    /// The name of the target of the directive.
-    pub fn target(&self) -> &str {
-        self.target
-    }
-}
-
-/// A trait encapsulating the operations required of a logger
-pub trait Log: Sync+Send {
-    /// Determines if a log message with the specified metadata would be
-    /// logged.
-    ///
-    /// This is used by the `log_enabled!` macro to allow callers to avoid
-    /// expensive computation of log message arguments if the message would be
-    /// discarded anyway.
-    fn enabled(&self, metadata: &LogMetadata) -> bool;
-
-    /// Logs the `LogRecord`.
-    ///
-    /// Note that `enabled` is *not* necessarily called before this method.
-    /// Implementations of `log` should perform all necessary filtering
-    /// internally.
-    fn log(&self, record: &LogRecord);
-}
-
-/// The location of a log message.
-///
-/// # Warning
-///
-/// The fields of this struct are public so that they may be initialized by the
-/// `log!` macro. They are subject to change at any time and should never be
-/// accessed directly.
-#[derive(Copy, Clone, Debug)]
-pub struct LogLocation {
-    #[doc(hidden)]
-    pub __module_path: &'static str,
-    #[doc(hidden)]
-    pub __file: &'static str,
-    #[doc(hidden)]
-    pub __line: u32,
-}
-
-impl LogLocation {
-    /// The module path of the message.
-    pub fn module_path(&self) -> &str {
-        self.__module_path
-    }
-
-    /// The source file containing the message.
-    pub fn file(&self) -> &str {
-        self.__file
-    }
-
-    /// The line containing the message.
-    pub fn line(&self) -> u32 {
-        self.__line
-    }
-}
-
-/// A token providing read and write access to the global maximum log level
-/// filter.
-///
-/// The maximum log level is used as an optimization to avoid evaluating log
-/// messages that will be ignored by the logger. Any message with a level
-/// higher than the maximum log level filter will be ignored. A logger should
-/// make sure to keep the maximum log level filter in sync with its current
-/// configuration.
-#[allow(missing_copy_implementations)]
-pub struct MaxLogLevelFilter(());
-
-impl fmt::Debug for MaxLogLevelFilter {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        write!(fmt, "MaxLogLevelFilter")
-    }
-}
+extern crate log_core;
 
-impl MaxLogLevelFilter {
-    /// Gets the current maximum log level filter.
-    pub fn get(&self) -> LogLevelFilter {
-        max_log_level()
-    }
+// Re-export most of log_core, but not set_logger_raw and shutdown_logger
+pub use log_core::{LogLevel, LogLevelFilter, LogRecord, LogMetadata, Log,
+                   LogLocation, MaxLogLevelFilter, max_log_level, __enabled,
+                   __log, __static_max_level};
 
-    /// Sets the maximum log level.
-    pub fn set(&self, level: LogLevelFilter) {
-        MAX_LOG_LEVEL_FILTER.store(level as usize, Ordering::SeqCst)
-    }
-}
+#[macro_use]
+#[path = "../core/src/macros.rs"]
+mod macros;
```

Comment:
`#[macro_reexport]` exists but it is unstable. I think this is because of issues with how `$crate` is handled.